File: angular/src/app/shared/models/receipt-detail.ts
import { BaseModel } from './base.model';

export class ReceiptDetail extends BaseModel {
  receiptId: number;
  productId: number;
  productName?: string;
  discountUnitPrice: number;
  unitPrice: number;
  quantity: number;

  constructor(
    id: number = 0,
    receiptId: number = 0,
    productId: number = 0,
    discountUnitPrice: number = 0,
    unitPrice: number = 0,
    quantity: number = 0,
    productName?: string
  ) {
    super(id);
    this.receiptId = receiptId;
    this.productId = productId;
    this.discountUnitPrice = discountUnitPrice;
    this.unitPrice = unitPrice;
    this.quantity = quantity;
    this.productName = productName;
  }
}

File: Abstraction/Entities/ReceiptDetail.cs
using Microsoft.EntityFrameworkCore;
using System.ComponentModel.DataAnnotations.Schema;

namespace Abstraction.Entities;

[Table("ReceiptDetail")]
[PrimaryKey(nameof(ReceiptId), nameof(ProductId))]
public class ReceiptDetail
    : BaseEntity
{
    public ReceiptDetail()
        : base()
    {
    }

    public ReceiptDetail(int id)
        : base(id)
    {
    }

    [ForeignKey(nameof(Receipt))]
    public int ReceiptId { get; set; }

    [ForeignKey(nameof(Product))]
    public int ProductId { get; set; }

    public decimal DiscountUnitPrice { get; set; }

    public decimal UnitPrice { get; set; }

    public int Quantity { get; set; }

    public virtual Receipt Receipt { get; set; }

    public virtual Product Product { get; set; }
}
